src/keboola_mcp_server/tools/components/validation.py

```diff
@@ -0,0 +1,145 @@
+from keboola_mcp_server.tools.components.utils import _get_component_details
+
+LOG = logging.getLogger(__name__)
+
+
+async def fetch_component_schema(
+        client: KeboolaClient, component_id: str
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """
+    Fetch configuration schemas directly from storage API for a specific component.
+
+    Args:
+        client: The Keboola client
+        component_id: The ID of the component
+
+    Returns:
+        A tuple of (root_configuration_schema, row_configuration_schema)
+    """
+    try:
+        endpoint = 'v2/storage/components'
+        params = {'include': component_id}
+
+        response = await client.storage_client.get(endpoint, params=params)
+
+        if not response or not isinstance(response, list) or len(response) == 0:
+            LOG.warning(f"No component found for ID {component_id}")
+            return None, None
+
+        component = response[0]
+        if component['id'] != component_id:
+            LOG.warning(f"Retrieved component ID {component['id']} doesn't match requested ID {component_id}")
+            return None, None
+
+        return component.get('configurationSchema'), component.get('configurationRowSchema')
+
+    except Exception as e:
+        LOG.exception(f"Error fetching schema for component {component_id}: {e}")
+        return None, None
+
+
+def validate_against_schema(configuration: Dict[str, Any], schema: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Validate a configuration against a JSON schema.
+
+    Args:
+        configuration: The configuration to validate
+        schema: The JSON schema to validate against
+
+    Returns:
+        Dictionary with validation results including 'valid' flag and any errors
+    """
+    try:
+        validator = Draft7Validator(schema)
+        errors = list(validator.iter_errors(configuration))
+
+        if not errors:
+            return {"valid": True, "errors": []}
+
+        formatted_errors = []
+        for error in errors:
+            path = ".".join(str(p) for p in error.path) if error.path else "root"
+            formatted_errors.append({
+                "path": path,
+                "message": error.message,
+                "schema_path": ".".join(str(p) for p in error.schema_path)
+            })
+
+        return {"valid": False, "errors": formatted_errors}
+
+    except exceptions.SchemaError as e:
+        return {
+            "valid": False,
+            "errors": [f"Schema error: {str(e)}"]
+        }
+    except Exception as e:
+        return {
+            "valid": False,
+            "errors": [f"Unexpected error: {str(e)}"]
+        }
+
+
+async def validate_component_configuration(
+        ctx: Context,
+        component_id: Annotated[
+            str,
+            Field(
+                description='The ID of the component to validate the configuration against.',
+            ),
+        ],
+        configuration: Annotated[
```

Comment:
we call this `parameters` in all the related tools (update/create). It may not matter but I would incline to keep it consistent.